src/streaming/multiplex/advanced.rs

```diff
@@ -304,7 +304,9 @@ impl<T> Multiplex<T> where T: Dispatch {
         while self.run {
             // TODO: Only read frames if there is available space in the frame
             // buffer
-            if let Async::Ready(frame) = try!(self.dispatch.get_mut().inner.transport().poll()) {
+            if self.frame_buf.full() {
```

Comment:
As I suspected, writing a small test that can consistently trigger this edge case is far from trivial. I'm in the middle of finals right now so I will not be able to get around to writing a unit test for this for a week or so